### DISCUSSION TITLE
Best deepmd-kit installation sequence?

### DISCUSSION THREAD

**Discussion by Mskater** [deepmodeling/deepmd-kit#260] [Q&A]
Can anyone please tell me the current, best option for installing deepmd-kit with the easiest install methods, and proper versions of tensorflow, bazel, LAMMPS, CUDA, etc.?   I am new to this as a grad student and have tried everything I've seen online for weeks.  Issues are version incompatibilities, dependencies, libraries not found, errors of all kinds. I don't want to enumerate specific issues, I'm just looking for someone who has had recent success with a route that I can try.
Thank you for any time!
I have:
Ubuntu 16.04
Python 3.5.2, 3.7.7 
pip 20.2.3,  20.1.1
gcc 7.5
miniconda3
jdk 9

**Comment by jameswind** [comment]
Does conda installation works for you? See here:
https://github.com/deepmodeling/deepmd-kit#with-conda

**Comment by Mskater** [comment]
@jameswind   Thanks for responding.   Yes, the conda installation works in itself.  Apparently it downloads TF 2.2 and CUDA 10.1.243.  These are apparently not compatible (ERRORs in trying to find 10.1 files), someone said CUDA 10.1 is compatible, but when trying to install that, it refuses (from official website).
Another issue is that tensorflow 2.2 will not build (./configure).   Deepmd-kit instructions say to use bazel 0.24.1.  My suspicion is that it too is not compatible with the new TF 2.2.  I had TF built another way before ((2.1) with bazel 0.19.1 installed), but when trying to build deepmd-kit after that, it could not find the TF files..( NOT FOUND "tensorflow/core/public/session.h" )    Also there is an issue with GPU support vs. CPU.  Deepmd-kit instructions say the gpu install is optional, and when I did not get success with CPU, I read on the Tensorflow blog that it is NOT optional?
So I'm going around in circles trying to find a compatible set of all the above software that will work together.
My hope is that someone on any Ubuntu system can tell the exact versions that worked for them, with the install methods used. That's all.  Thanks!

**Comment by PabloPiaggi** [comment]
Hi @Mskater , these are the instructions I use in one of our clusters:

https://github.com/CSIprinceton/CSI-hacks-and-tricks/tree/master/Compilation/DeePMD#tigergpu

I don't claim they are the best nor the easiest, but you might find them useful.

**Comment by Mskater** [comment]
@PabloPiaggi   Thanks, I will certainly try.

**Comment by njzjz** [comment]
We have not tested deepmd-kit C++ interface with TensorFlow 2.2.  The latest version is TensorFlow 2.1. For the build step, see the following codes:
* https://github.com/deepmd-kit-recipes/libtensorflow_cc-feedstock/blob/1ddd67457fe5486f822094b95aaea5326982781c/recipe/build_gpu.sh#L1-L99
* https://github.com/deepmd-kit-recipes/libdeepmd-feedstock/blob/a7c7bfec03b1d0a363e25067b68fb4a5d7857ce2/recipe/build.sh#L1-L9
* https://github.com/deepmd-kit-recipes/lammps-dp-feedstock/blob/e2fecdf255ac739d01247ce815944ead3b91e9a4/recipe/build.sh#L1-L17.
